src/vm/runtimecallablewrapper.cpp

```diff
@@ -131,127 +129,115 @@ IUnknown *ComClassFactory::CreateInstanceFromClassFactory(IClassFactory *pClassF
     }
     else
     {
-        if (m_pClassMT == NULL)
-        {
-            // Create an instance of the object.
-            FrameWithCookie<DebuggerExitFrame> __def;
-            {
-                GCX_PREEMP();
-                hr = pClassFact->CreateInstance(punkOuter, IID_IUnknown, (void **)&pUnk);
-                if (FAILED(hr) && punkOuter)
-                {
-                    hr = pClassFact->CreateInstance(NULL, IID_IUnknown, (void**)&pUnk);
-                    if (pfDidContainment)
-                        *pfDidContainment = TRUE;
-                }
-            }
-            __def.Pop();
-        }
-        else
+        _ASSERTE(m_pClassMT != NULL);
+
+        // Get the type to query for licensing.
+        TypeHandle rth = TypeHandle(m_pClassMT);
+
+        MethodTable *pProxyMT = MscorlibBinder::GetClass(CLASS__LICENSE_INTEROP_PROXY);
+        MethodDesc *pMD = MemberLoader::FindMethod(pProxyMT, "Create", &gsig_SM_RetObj);
```

Comment:
It is better to reference the method using `METHOD__XXX` instead of name. It will prevent the method to be trimmed from CoreLib among other things.

Reply:
I can, but that makes this sequence a bit different. Do I use `MscorlibBinder` with the Binder ID? Just unclear on how to use the Binder ID rather than the name.

Reply:
Just pass the ID to MethodDescCallSite constructor. Most of the uses of MethodDescCallSite do just that. One example from many:
```
            MethodDescCallSite getMessage(METHOD__EXCEPTION__TO_STRING, &objException);
```

Reply:
So much better. Thanks!